designs/tinkerbell-provider-input.md

```diff
@@ -0,0 +1,358 @@
+  * Secret.data.username: Base64 encoded username for BMC
+  * Secret.data.password: Base64 encoded password for BMC
+  * Secret.type: Type of secret which should be set to `kubernetes.io/basic-auth`
+
+Here is an example Hardware YAML for one hardware
+```yaml
+apiVersion: tinkerbell.org/v1alpha1
+kind: Hardware
+metadata:
+  creationTimestamp: null
+  labels:
+    clusterctl.cluster.x-k8s.io/move: "true"
+  name: worker1
+  namespace: eksa-system
+spec:
+  bmcRef: bmc-worker1
+  id: 16035822-7813-42b2-ae9a-04eae8574a73
+status: {}
+---
+apiVersion: tinkerbell.org/v1alpha1
+kind: BMC
+metadata:
+  creationTimestamp: null
+  labels:
+    clusterctl.cluster.x-k8s.io/move: "true"
+  name: bmc-worker1
+  namespace: eksa-system
+spec:
+  authSecretRef:
+    name: bmc-worker1-auth
+    namespace: eksa-system
+  host: 192.168.0.10
+  vendor: supermicro
+status: {}
+---
+apiVersion: v1
+data:
+  password: QWRtaW4=
+  username: YWRtaW4=
+kind: Secret
+metadata:
+  creationTimestamp: null
+  labels:
+    clusterctl.cluster.x-k8s.io/move: "true"
+  name: bmc-worker1-auth
+  namespace: eksa-system
+type: kubernetes.io/basic-auth
+---
+```
+
+**Generating Hardware Manifests**
+
+Users need to generate and maintain one copy of hardware JSON and hardware YAML for each hardware they want to use with EKS Anywhere. This can get tedious. So, we are adding a new subcommand under `eksctl anywhere generate` to make this easier.
+
+This new command will take a CSV file as input with configurations for all the hardware, parse it and output the required hardware manifests. This will also by default push the hardware to the Tinkerbell stack.
+
+The CSV file will need these headers to identify which field corresponds to which value:
+* ip_address: IP address to configure on the machine
+* gateway: Gateway to configure on the machine
+* nameservers: List of nameservers to be specified on the machine, delimited by `|`
+* netmask: Netmask for the IP
+* mac: MAC address of the interface to assign the IP to
+* hostname: Hostname of the machine
+* vendor: Name of the machine's vendor
+* bmc_ip: IP address of the machine's BMC interface
+* bmc_username: Username for the machine's BMC
+* bmc_password: Password for machine's BMC
+
+All of these fields are required and can be placed in any order in the CSV file.
+
+Here is an example hardware CSV file with the required information for two hardwares:
+```csv
+ip_address,gateway,nameservers,netmask,mac,hostname,vendor,bmc_ip,bmc_username,bmc_password
+192.168.1.10,192.168.0.1,8.8.8.8|8.8.4.4,255.255.255.0,00:00:00:00:00:00,worker1,supermicro,192.168.0.10,admin,Admin
+192.168.1.11,192.168.0.1,8.8.8.8|8.8.4.4,255.255.255.0,00:00:00:00:00:01,worker2,supermicro,192.168.0.11,admin,Admin
+```
+
+Once the CSV file is set up, users can use the `generate hardware` command to generate the hardware manifests
+
+(Recommended) Command to generate the hardware manifests and push it to the Tinkerbell stack
```

Comment:
not blocking: tbh I don't really like this. I think it breaks the pattern of the `generate` command, where it doesn't have side effects, it just writes files to local disk and lets the user decide what to do with those files.

My vote would be to make "not pushing" the default. Or even better, move the "push" functionality to a separate command.

Reply:
I agree with `generate` and `push` being separate commands, or with a flag. Maybe we have a `setup` that does both

Reply:
The reason we wanted to push hardware by default as part of generate is mainly that it is tedious to push hardware manually to Tinkerbell. We also wanted to keep it part of the same command so users wouldn't have to run two different commands that can be part of just one. 

This is also just a beta thing and most likely the push part will go away at GA (still have to finalize a few details for GA)
If you guys prefer, I could move it to a separate `setup` command like Vivek suggested. Let me know what you guys think.

Reply:
If it's not a lot of work, I like the separate command idea

Reply:
I have updated the doc to use `setup` instead of `generate`